src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -6341,9 +6349,19 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
             } else result
           }
 
-        val result =
-          if (shouldInsertStabilizers) addStabilizers(context.pendingStabilizers, adapted)
-          else adapted
+        def isRassoc(t: Tree) = t match {
+          case ValDef(_, nm, _, _) => nm.startsWith(nme.RIGHT_ASSOC_OP_PREFIX)
+          case _ => false
+        }
+        val shouldInsertRassocs = context.pendingStabilizers.exists(isRassoc)
+        // if in the midst of series of apply, possibly a qualifier was adapted, possibly an rassoc arg.
+        // the expression to rewrite with added stabilizers is different for each case.
+        val result = adapted match {
+          case view @ Apply(coercion, coerced :: Nil) if view.hasAttachment[AppliedImplicitView.type] && shouldInsertRassocs =>
+            treeCopy.Apply(view, coercion, addStabilizers(coerced) :: Nil)
```

Comment:
this piece made me wonder.

so the example is this

```
class L
object E extends L
class C(val h: Int, val t: L) extends L

class D(val l: () => L) {
  def #:: (o: Int): L = new C(o, l())
}

object T {
  implicit def toD(l: => L): D = new D(() => l)
}

a #:: b #:: E
```



When the outer implicit conversion is typed (adaptToMember, `T.toD(T.toD(E).#::(rassoc$1))`)

   - we end up in `typedApply` as expected. The argument `T.toD(E).#::(rassoc$1)` is typed without `APPSELmode`, that flag is removed in `typedArg`.
   - but the argument tree is already typed, and importantly, `context.pendingStabilizers` is already populated with `rassoc$1`
   - at the beginning of `def typed`
     - `shouldInsertStabilizers` is true, we save the pending one in a local and set `context.pendingStabilizers = Nil`
     - this means the stabilizer is not added by mistake. the logic around that in `def typed` doesn't work with trees that are already typed

we can probably fix that instead of the special case here?

Reply:
There is no emoji reaction with blurry eyes, but I'll think about it later.